app/frontend/src/components/Comments/CommentBox.tsx

```diff
@@ -0,0 +1,118 @@
+import { Card, makeStyles } from "@material-ui/core";
+import React, { useEffect, useState } from "react";
+
+import Alert from "../../components/Alert";
+import CircularProgress from "../../components/CircularProgress";
+import Markdown from "../../components/Markdown";
+import { Reply } from "../../pb/threads_pb";
+import { service } from "../../service";
+import NewComment from "./NewComment";
+
+const useStyles = makeStyles((theme) => ({
+  card: {
+    marginTop: "1em",
+    padding: "1em",
+    border: "1px solid",
+  },
+}));
+
+interface CommentBoxProps {
+  threadId: number;
+}
+
+// Reply with more Reply objects as children
+interface MultiLevelReply extends Reply.AsObject {
+  replies: Array<Reply.AsObject>;
+  // page token, etc? not sure what's needed for react query
+}
+
+export default function CommentBox({ threadId }: CommentBoxProps) {
+  const classes = useStyles();
+
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState("");
+
+  const [comments, setComments] = useState<Array<MultiLevelReply>>([]);
+
+  useEffect(() => {
+    (async () => {
+      setLoading(true);
+      try {
+        const thread = await service.threads.getThread(threadId);
+        setComments(
```

Comment:
Probably ok for now... but definitely should do this in React Query to avoid having to deal with all the refetching ourselves. Is this meant to be working with pagination as well already?

Reply:
> Is this meant to be working with pagination as well already?

Seems not, that's easier with react-query anyway